src/zigbee2mqtt/zigbee2mqtt-adapter.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ *
+ * ZigbeeClassifier - Determines properties from Zigbee clusters.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+import { Adapter, AddonManagerProxy, Device } from 'gateway-addon';
+import { Config, Zigbee2MQTTAdapter } from '../config';
+import mqtt from 'mqtt';
+import { Zigbee2MqttDevice } from './zigbee2mqtt-device';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const manifest = require('../../manifest.json');
+
+interface Response {
+  data?: {
+    id?: string;
+    block?: boolean;
+    force?: boolean;
+    value: boolean;
+  };
+  status?: string;
+  error?: string;
+}
+
+interface Log {
+  level?: string;
+  message?: string;
+}
+
+const DEVICES_POSTFIX = '/bridge/devices';
+const PERMIT_REQUEST_POSTFIX = '/bridge/request/permit_join';
+const PERMIT_RESPONSE_POSTFIX = '/bridge/response/permit_join';
+const REMOVE_REQUEST_POSTFIX = '/bridge/request/device/remove';
+const REMOVE_RESPONSE_POSTFIX = '/bridge/response/device/remove';
+const LOGGING_POSTFIX = '/bridge/logging';
+
+export class Zigbee2MqttAdapter extends Adapter {
+  private prefix: string;
+
+  private debug: boolean;
+
+  private client?: mqtt.Client;
+
+  constructor(
+    addonManager: AddonManagerProxy,
+    private config: Config,
+    private adapterConfig: Zigbee2MQTTAdapter
+  ) {
+    super(addonManager, manifest.id, manifest.id);
```

Comment:
Since there can be more than one of these adapters, the second parameter of the `super()` call needs to be something unique, like maybe:
```typescript
super(addonManager, `zb-zigbee2mqtt-${adapterConfig.host}:${adapterConfig.port}`, manifest.id);
```

Reply:
done